packages/alfa-rules/src/sia-r70/rule.ts

```diff
@@ -0,0 +1,137 @@
+import { Rule, Diagnostic } from "@siteimprove/alfa-act";
+import { Document, Element, Namespace } from "@siteimprove/alfa-dom";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Err, Ok } from "@siteimprove/alfa-result";
+import { Page } from "@siteimprove/alfa-web";
+import { List } from "@siteimprove/alfa-list";
+import { Iterable } from "@siteimprove/alfa-iterable";
+import { Array } from "@siteimprove/alfa-array";
+
+import { expectation } from "../common/expectation";
+import { hasChild, isRendered, isDocumentElement } from "../common/predicate";
+
+const { isElement, hasName, hasNamespace } = Element;
+const { and, test } = Predicate;
+
+const isDeprecated = hasName(
+  "acronym",
+  "applet",
+  "basefont",
+  "bgsound",
+  "big",
+  "blink",
+  "center",
+  "content",
+  "dir",
+  "font",
+  "frame",
+  "frameset",
+  "hgroup",
+  "image",
+  "keygen",
+  "marquee",
+  "menuitem",
+  "nobr",
+  "noembed",
+  "noframes",
+  "plaintext",
+  "rb",
+  "rtc",
+  "shadow",
+  "spacer",
+  "strike",
+  "tt",
+  "xmp"
+);
+
+export default Rule.Atomic.of<Page, Document>({
+  uri: "https://alfa.siteimprove.com/rules/sia-r70",
+  evaluate({ device, document }) {
+    return {
+      applicability() {
+        return test(hasChild(isDocumentElement), document) ? [document] : [];
+      },
+
+      expectations(target) {
+        const deprecatedElements = target
+          .descendants({ flattened: true, nested: true })
+          .filter(isElement)
+          .filter(
+            and(hasNamespace(Namespace.HTML), isDeprecated, isRendered(device))
+          );
+
+        return {
+          1: expectation(
+            deprecatedElements.isEmpty(),
+            () => Outcomes.HasNoDeprecatedElement,
+            () => Outcomes.HasDeprecatedElements(deprecatedElements)
+          ),
+        };
+      },
+    };
+  },
+});
+
+export namespace Outcomes {
+  export const HasNoDeprecatedElement = Ok.of(
+    Diagnostic.of(`The document doesn't contain any deprecated element`)
+  );
+
+  export const HasDeprecatedElements = (errors: Iterable<Element>) =>
+    Err.of(
+      DeprecatedElements.of(
+        `The document contains deprecated elements: `,
```

Comment:
```suggestion
        `The document contains deprecated elements`,
```

Reply:
We have that `errors` is an `Iterable<Element>`, so isn´t it more readable to leave : before the list is printed out? 
```
DeprecatedElements.of(
        `The document contains deprecated elements: `,
        errors
      )
```

Reply:
The message is used independently of the diagnostic data and so the two won't necessarily be output together.